tests/end_to_end/end_to_end.cpp

```diff
@@ -376,26 +390,26 @@ droppacket_test(ebpf_execution_type_t execution_type)
 
     int result;
     const char* error_message = nullptr;
-    bpf_object* object = nullptr;
+    bpf_object_ptr unique_object;
     fd_t program_fd;
-    bpf_link* link;
+    bpf_link_ptr link;
 
     single_instance_hook_t hook(EBPF_PROGRAM_TYPE_XDP, EBPF_ATTACH_TYPE_XDP);
     program_info_provider_t xdp_program_info(EBPF_PROGRAM_TYPE_XDP);
 
     const char* file_name = (execution_type == EBPF_EXECUTION_NATIVE ? "droppacket_um.dll" : "droppacket.o");
-
-    result = ebpf_program_load(file_name, BPF_PROG_TYPE_UNSPEC, execution_type, &object, &program_fd, &error_message);
+    result =
+        ebpf_program_load(file_name, BPF_PROG_TYPE_UNSPEC, execution_type, &unique_object, &program_fd, &error_message);
```

Comment:
If we revisit this call's implementation in future, could we should consider converting the return value type to that of an ```std::variant``` of the error code and 'unique_object'?  We could also convert the 'error_message' argument to an ```std::optional```.

(The current implementation is fine for now)